{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64e1acba",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[I 2025-06-26 23:51:01,609] A new study created in memory with name: no-name-1ffa31c0-ab15-4f14-9471-da74519d041a\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,616] Trial 0 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 28, 'num_layers': 2, 'dropout': 0.10150396655160426, 'lr': 0.0006050451587709993, 'batch_size': 48}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,620] Trial 1 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 39, 'num_layers': 2, 'dropout': 0.014462779538141401, 'lr': 0.0021737320100735443, 'batch_size': 112}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,625] Trial 2 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 113, 'num_layers': 3, 'dropout': 0.012081071325605375, 'lr': 0.0009573628595355673, 'batch_size': 80}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,629] Trial 3 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 35, 'num_layers': 2, 'dropout': 0.42622357173021536, 'lr': 0.008018565518814275, 'batch_size': 128}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,634] Trial 4 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 40, 'num_layers': 2, 'dropout': 0.1768462445622775, 'lr': 0.0011300085747253598, 'batch_size': 128}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,639] Trial 5 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 33, 'num_layers': 1, 'dropout': 0.13195258324197245, 'lr': 0.0011689389583861114, 'batch_size': 16}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,644] Trial 6 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 110, 'num_layers': 1, 'dropout': 0.22553950499162123, 'lr': 0.0006592056489279585, 'batch_size': 80}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,648] Trial 7 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 106, 'num_layers': 2, 'dropout': 0.15664513691659504, 'lr': 0.0001712688944415087, 'batch_size': 112}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,652] Trial 8 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 49, 'num_layers': 3, 'dropout': 0.3570949333401869, 'lr': 0.00011523233427278905, 'batch_size': 80}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,657] Trial 9 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 26, 'num_layers': 2, 'dropout': 0.005853146926799391, 'lr': 0.00016994464900369218, 'batch_size': 16}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,673] Trial 10 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 75, 'num_layers': 1, 'dropout': 0.2965464719409584, 'lr': 0.0004641223787043194, 'batch_size': 48}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,687] Trial 11 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 63, 'num_layers': 2, 'dropout': 0.07586646223693595, 'lr': 0.0035206656906637177, 'batch_size': 64}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,701] Trial 12 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 19, 'num_layers': 3, 'dropout': 0.0841499853817104, 'lr': 0.0024266959551389896, 'batch_size': 48}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,714] Trial 13 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 59, 'num_layers': 2, 'dropout': 0.07288641591452721, 'lr': 0.002552513079148406, 'batch_size': 96}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,726] Trial 14 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 81, 'num_layers': 1, 'dropout': 0.2572152346947183, 'lr': 0.00039781905776683547, 'batch_size': 48}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,738] Trial 15 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 48, 'num_layers': 2, 'dropout': 0.013688678830350895, 'lr': 0.008976548460416492, 'batch_size': 112}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,751] Trial 16 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 89, 'num_layers': 3, 'dropout': 0.1121931055402905, 'lr': 0.004520901254431675, 'batch_size': 32}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,763] Trial 17 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 23, 'num_layers': 2, 'dropout': 0.2130659058896403, 'lr': 0.0016702269443991079, 'batch_size': 64}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,776] Trial 18 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 58, 'num_layers': 1, 'dropout': 0.05186535639488171, 'lr': 0.0002960662119226748, 'batch_size': 96}. Best is trial 0 with value: inf.\n",
      "C:\\Users\\usuario\\AppData\\Local\\Temp\\ipykernel_19956\\1073141642.py:99: FutureWarning: suggest_int() got {'step'} as positional arguments but they were expected to be given as keyword arguments.\n",
      "  bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "[I 2025-06-26 23:51:01,789] Trial 19 finished with value: inf and parameters: {'seq_len': 368, 'hidden_dim': 16, 'num_layers': 3, 'dropout': 0.31212775047747915, 'lr': 0.0006755474230008847, 'batch_size': 32}. Best is trial 0 with value: inf.\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n",
      "c:\\Git\\ICESI\\FA_TALLER2\\venv\\Lib\\site-packages\\sklearn\\utils\\validation.py:2739: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "empty dataset",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mValueError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 116\u001b[39m\n\u001b[32m    113\u001b[39m study.optimize(objective, n_trials=\u001b[32m20\u001b[39m)\n\u001b[32m    114\u001b[39m best = study.best_params\n\u001b[32m--> \u001b[39m\u001b[32m116\u001b[39m train_loader, scaler = \u001b[43mget_loader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdf_train\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbest\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mseq_len\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbest\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mbatch_size\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfit_scaler\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[32m    117\u001b[39m model = GRUModel(best[\u001b[33m'\u001b[39m\u001b[33mhidden_dim\u001b[39m\u001b[33m'\u001b[39m], best[\u001b[33m'\u001b[39m\u001b[33mnum_layers\u001b[39m\u001b[33m'\u001b[39m], best[\u001b[33m'\u001b[39m\u001b[33mdropout\u001b[39m\u001b[33m'\u001b[39m]).to(DEVICE)\n\u001b[32m    118\u001b[39m opt = torch.optim.Adam(model.parameters(), lr=best[\u001b[33m'\u001b[39m\u001b[33mlr\u001b[39m\u001b[33m'\u001b[39m])\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[1]\u001b[39m\u001b[32m, line 50\u001b[39m, in \u001b[36mget_loader\u001b[39m\u001b[34m(df, seq_len, batch_size, shuffle, scaler, fit_scaler)\u001b[39m\n\u001b[32m     48\u001b[39m ds = DengueDataset(df, seq_len, scaler, fit_scaler)\n\u001b[32m     49\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(ds) == \u001b[32m0\u001b[39m:\n\u001b[32m---> \u001b[39m\u001b[32m50\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[33m\"\u001b[39m\u001b[33mempty dataset\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m     51\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m DataLoader(ds, batch_size=batch_size, shuffle=shuffle), ds.scaler\n",
      "\u001b[31mValueError\u001b[39m: empty dataset"
     ]
    }
   ],
   "source": [
    "import random\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import torch\n",
    "import torch.nn as nn\n",
    "from torch.utils.data import Dataset, DataLoader\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "import optuna\n",
    "\n",
    "SEED = 42\n",
    "random.seed(SEED)\n",
    "np.random.seed(SEED)\n",
    "torch.manual_seed(SEED)\n",
    "if torch.cuda.is_available():\n",
    "    torch.cuda.manual_seed_all(SEED)\n",
    "\n",
    "DEVICE = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
    "\n",
    "train_df = pd.read_parquet('../../Datos/df_train.parquet')\n",
    "test_df = pd.read_parquet('../../Datos/df_test.parquet')\n",
    "\n",
    "lengths = train_df.groupby('id_bar').size()\n",
    "min_len = int(lengths.quantile(0.05))\n",
    "max_len = int(lengths.quantile(0.95))\n",
    "\n",
    "bars = train_df['id_bar'].unique()\n",
    "train_bars, val_bars = train_test_split(bars, test_size=0.2, random_state=SEED)\n",
    "df_train = train_df[train_df['id_bar'].isin(train_bars)].reset_index(drop=True)\n",
    "df_val = train_df[train_df['id_bar'].isin(val_bars)].reset_index(drop=True)\n",
    "\n",
    "class DengueDataset(Dataset):\n",
    "    def __init__(self, df, seq_len, scaler=None, fit_scaler=False):\n",
    "        self.seq_len = seq_len\n",
    "        self.scaler = StandardScaler().fit(df[['dengue']]) if fit_scaler else scaler\n",
    "        self.data = []\n",
    "        for _, grp in df.groupby('id_bar'):\n",
    "            values = self.scaler.transform(grp[['dengue']].values)\n",
    "            for i in range(len(values) - seq_len):\n",
    "                self.data.append((values[i:i+seq_len].astype(np.float32), values[i+seq_len].astype(np.float32)))\n",
    "    def __len__(self):\n",
    "        return len(self.data)\n",
    "    def __getitem__(self, idx):\n",
    "        x, y = self.data[idx]\n",
    "        return x, y\n",
    "\n",
    "def get_loader(df, seq_len, batch_size, shuffle, scaler=None, fit_scaler=False):\n",
    "    ds = DengueDataset(df, seq_len, scaler, fit_scaler)\n",
    "    if len(ds) == 0:\n",
    "        raise ValueError(\"empty dataset\")\n",
    "    return DataLoader(ds, batch_size=batch_size, shuffle=shuffle), ds.scaler\n",
    "\n",
    "class GRUModel(nn.Module):\n",
    "    def __init__(self, hidden_dim, num_layers, dropout):\n",
    "        super().__init__()\n",
    "        self.gru = nn.GRU(\n",
    "            input_size=1,\n",
    "            hidden_size=hidden_dim,\n",
    "            num_layers=num_layers,\n",
    "            dropout=dropout if num_layers>1 else 0,\n",
    "            batch_first=True\n",
    "        )\n",
    "        self.norm = nn.LayerNorm(hidden_dim)\n",
    "        self.fc = nn.Linear(hidden_dim, 1)\n",
    "    def forward(self, x):\n",
    "        out, _ = self.gru(x)\n",
    "        h = self.norm(out[:, -1, :])\n",
    "        return self.fc(h)\n",
    "\n",
    "def train_epoch(model, loader, criterion, optimizer):\n",
    "    model.train()\n",
    "    total = 0\n",
    "    for x, y in loader:\n",
    "        x = x.unsqueeze(-1).to(DEVICE)\n",
    "        y = y.to(DEVICE)\n",
    "        optimizer.zero_grad()\n",
    "        loss = criterion(model(x).squeeze(), y.squeeze())\n",
    "        loss.backward()\n",
    "        optimizer.step()\n",
    "        total += loss.item() * x.size(0)\n",
    "    return total / len(loader.dataset)\n",
    "\n",
    "def validate(model, loader, criterion):\n",
    "    model.eval()\n",
    "    total = 0\n",
    "    with torch.no_grad():\n",
    "        for x, y in loader:\n",
    "            x = x.unsqueeze(-1).to(DEVICE)\n",
    "            y = y.to(DEVICE)\n",
    "            total += criterion(model(x).squeeze(), y.squeeze()).item() * x.size(0)\n",
    "    return total / len(loader.dataset)\n",
    "\n",
    "def objective(trial):\n",
    "    seq_len = trial.suggest_int('seq_len', min_len, max_len)\n",
    "    hd = trial.suggest_int('hidden_dim', 16, 128)\n",
    "    nl = trial.suggest_int('num_layers', 1, 3)\n",
    "    do = trial.suggest_float('dropout', 0.0, 0.5)\n",
    "    lr = trial.suggest_float('lr', 1e-4, 1e-2, log=True)\n",
    "    bs = trial.suggest_int('batch_size', 16, 128, 16)\n",
    "    try:\n",
    "        train_loader, scaler = get_loader(df_train, seq_len, bs, True, fit_scaler=True)\n",
    "        val_loader, _ = get_loader(df_val, seq_len, bs, False, scaler)\n",
    "    except ValueError:\n",
    "        return float('inf')\n",
    "    model = GRUModel(hd, nl, do).to(DEVICE)\n",
    "    opt = torch.optim.Adam(model.parameters(), lr=lr)\n",
    "    crit = nn.MSELoss()\n",
    "    for _ in range(10):\n",
    "        train_epoch(model, train_loader, crit, opt)\n",
    "    return validate(model, val_loader, crit)\n",
    "\n",
    "study = optuna.create_study(direction='minimize')\n",
    "study.optimize(objective, n_trials=20)\n",
    "best = study.best_params\n",
    "\n",
    "train_loader, scaler = get_loader(df_train, best['seq_len'], best['batch_size'], True, fit_scaler=True)\n",
    "model = GRUModel(best['hidden_dim'], best['num_layers'], best['dropout']).to(DEVICE)\n",
    "opt = torch.optim.Adam(model.parameters(), lr=best['lr'])\n",
    "crit = nn.MSELoss()\n",
    "for epoch in range(1, 51):\n",
    "    loss = train_epoch(model, train_loader, crit, opt)\n",
    "    if epoch % 10 == 0:\n",
    "        print(epoch, loss)\n",
    "\n",
    "predictions = []\n",
    "for bar_id, grp in test_df.groupby('id_bar'):\n",
    "    history = train_df[train_df['id_bar']==bar_id]['dengue'].values\n",
    "    seq = scaler.transform(history.reshape(-1,1)).flatten().tolist()\n",
    "    for _ in grp.itertuples():\n",
    "        window = np.array(seq[-best['seq_len']:]).reshape(1, best['seq_len'], 1).astype(np.float32)\n",
    "        p = model(torch.tensor(window).to(DEVICE)).cpu().item()\n",
    "        seq.append(p)\n",
    "        inv = scaler.inverse_transform([[p]])[0][0]\n",
    "        predictions.append({'id_bar': bar_id, 'dengue': max(0, inv)})\n",
    "\n",
    "sub = pd.read_csv('../../Datos/sample_submission.csv')\n",
    "out = sub.drop(columns=['dengue']).merge(pd.DataFrame(predictions), on='id_bar', how='left')\n",
    "out.to_csv('predictions.csv', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
